starboard/shared/starboard/media/decoder_state_tracker.h

```diff
@@ -0,0 +1,77 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef STARBOARD_SHARED_STARBOARD_MEDIA_DECODER_STATE_TRACKER_H_
+#define STARBOARD_SHARED_STARBOARD_MEDIA_DECODER_STATE_TRACKER_H_
+
+#include <functional>
+#include <iosfwd>
+#include <map>
+#include <mutex>
+
+#include "starboard/shared/starboard/player/job_queue.h"
+
+namespace starboard {
+
+class DecoderStateTracker
+    : private shared::starboard::player::JobQueue::JobOwner {
+ public:
+  using StateChangedCB = std::function<void()>;
```

Comment:
std::function is banned in Chromium:
 https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++-features.md#std_function-banned

Luckily in this case we can just use `base::OnceCallback<void()>` or
`base::RepeatingCallback<void()>`.